### ISSUE TITLE
Missing 'note' or 'comment' field

### ISSUE THREAD

**Issue by shadow-ru** [MrLokans/MoonReader_tools#9]
Hi. Thanks for useful library!

There is 'note' or 'comment' field missing. I successfully fixed pip version and tried to create pull request here, but git version don't work. So that's my findings about file format:

```
CORRESP_TABLE = (
    (0, 'unknown_1'),
    (1, 'page'),
    (2, 'timestamp'),
    (3, 'unknown_2'),
    (4, 'unknown_3'),
    (5, 'color'),
    (6, 'style'),
    (7, 'note'), # user's note or comment to text's highlight
    (8, 'text'), # text's highlight
    (9, None)
)

NOTE_SCHEME = [
    # position, len, name
    (0, 1, 'note_id'),
    (1, 1, 'title'),
    (2, 1, 'path'),
    (3, 1, 'path_lower'),
    (4, 1, 'last_chapter'),
    (5, 1, 'last_split_index'),
    (6, 1, 'last_position'),
    (7, 1, 'highlight_length'),
    (8, 1, 'color'),  # number, that shows the color styling has
    (9, 1, 'timestamp'),  # integer, that shows when note was made
    (10, 1, 'separator_space'),
    (11, 1, 'note'), # user's note or comment to text's highlight
    (12, 1, 'text'),  # text's highlight
    (13, 3, 'modifier_bits'),  # is note deleted, e.g.
]
```
Hope this helps.

**Comment by MrLokans** [comment]
@shadow-rus, thanks for the desire to help? Could you please provide more information on where exactly are those fields missing? Perhaps a use-case example or a certain file causing those fields to be lost?

**Comment by shadow-ru** [comment]
You can add your own note (or comment) to text highlight, in that case it would be save to these fields:

```
CORRESP_TABLE = (
    (7, 'note'), # user's note or comment to text's highlight
 )

NOTE_SCHEME = [
    (11, 1, 'note'), # user's note or comment to text's highlight
]
```
IPython session:

```
>>>> from moonreader_tools.handlers import FilesystemDownloader
>>>> p = 'some_path'
>>>> handler = FilesystemDownloader()
>>>> books = handler.get_books(path=p)
>>>> b = books[0]
>>>>
>>>> b.to_dict()
>>>> {'notes': [{'date': '2017-08-26 22:52:45',
>>>>    'note': '',
>>>>    'text': 'Syntopicon--the first index of ideas in the\r\nhistory of the world.\r'},
>>>>   {'date': '2017-08-26 23:19:26',
>>>>    'note': '',
>>>>    'text': 'One of the most fascinating conclusions to come out of\r\nthis study was the observation that our complex Western\r\nCivilization revolves around only 102 great ideas. All other\r\nconcepts are subtropics of these great ideas (see back flap).'},
>>>>   {'date': '2017-10-15 12:00:22',
>>>>    'note': 'my note',
>>>>    'text': 'knowledge of the great books of history is\r\nunparalleled'}],
>>>>  'pages': 9,
>>>>  'percentage': 2.4,
>>>>  'title': 'Adler Mortimer J. - How to Read a Book (Special Edition)'}
```
Pay attention to 'note' field (`'note': 'my note'` in 3rd dict).

**Comment by MrLokans** [comment]
@shadow-ru, and what behavior do you expect in this case? Is it about incorrect naming? I could've renamed the 'text' field into 'highlight_text' and 'note' into 'user_note'

**Comment by shadow-ru** [comment]
Maybe I was not clear enough... Your library doesn't handle this field, I fix it by myself. Can you add support for this field, please?

Diff for pip version:

```
29c29,30
<                  path_lower=None
---
>                  path_lower=None,
>                  note=None
38a40
>         self.note = note
53c55,56
<             'date': str(date_from_long_timestamp(self._timestamp))
---
>             'date': str(date_from_long_timestamp(self._timestamp)),
>             'note': self.note,
75c78,80
<                    content="")
---
>                    content="",
>                    note=""
>                 )
93c98
<         (7, 'unknown_4'),
---
>         (7, 'note'),
110c115
<     def __init__(self, text, timestamp, style, color, content=""):
---
>     def __init__(self, text, timestamp, style, color, content="", note=""):
116c121,123
<                                       content=content)
---
>                                       content=content,
>                                       note=note
>                                     )
129c136,138
<                    content=text)
---
>                    content=text,
>                    note=token_dict.get("note", "")
>                 )
192,193c201,203
<         (10, 2, "separator_space"),
<         (12, 1, 'text'),  # actually, note's text
---
>         (10, 1, "separator_space"),
>         (11, 1, "note"),  # note
>         (12, 1, 'text'),  # highlight
205c215,217
<                  path_lower=None):
---
>                  path_lower=None,
>                  note=None
>                 ):
212c224,226
<                                       path_lower=path_lower)
---
>                                       path_lower=path_lower,
>                                       note=note
>                                     )
239c253,255
<                    path_lower=one_obj_or_list(book_dict['book_path_lower']))
---
>                    path_lower=one_obj_or_list(book_dict['book_path_lower']),
>                    note=one_obj_or_list(book_dict['note'])
>                 ) 
```

**Comment by MrLokans** [comment]
Thank you, will see what I can do with it

**Comment by MrLokans** [comment]
Actually, I'm going to rewrite the library so that it has a more obvious class hierarchy, and will include this fix in the upcoming update. Thank you for sharing the problem

**Comment by shadow-ru** [comment]
Ну, вроде разобрались. Спасибо!

**Comment by MrLokans** [comment]
@shadow-ru  Пожалуйста:)

**Comment by MrLokans** [comment]
@shadow-ru, fixed that in version 1.1.0 (available on PyPI already), thank you for your patience
